File: RMS/rms-service/src/main/java/com/maven/rms/models/OTCCtrBalPhy.java
package com.maven.rms.models;

import java.math.BigDecimal;
import java.math.BigInteger;
import java.util.Date;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class OTCCtrBalPhy {
    private String counter_id;
    
    private BigDecimal total_cash_amt;
    private String col_slip_no;
    private String orn_no;
    private String che_bank_nm;
    private String che_payer_nm;
    private String che_ba_acct_no;
    private String che_no;
    private Date che_date;
    private BigDecimal che_amt;
    private String bd_bank_nm;
    private String bd_no;
    private Date bd_date;
    private BigDecimal bd_amt;
    private String mo_rm_no;
    private Date mo_date;
    private String mo_payer_nm;
    private String mo_id_no;
    private String mo_contact_no;
    private BigDecimal mo_amt;
    private String detail_type;

    private BigInteger id;
    private BigInteger otc_id;

    private BigInteger otc_counter_id;

}

File: RMS-BackOfficePortal/src/app/core/models/otc-ctr-bal-phy.ts
export interface OTCPHYInfo{
    //2025-05-20 New added by Geo
    bal_date: Date;
    branch_code: String;
    
    total_cash_amt: number;       // Corresponds to BigDecimal in Java
    col_slip_no: string;
    orn_no: string;
    che_bank_nm: string;
    che_payer_nm: string;
    che_ba_acct_no: string;
    che_no: string;
    che_date: Date;               // Corresponds to Date in Java
    che_amt: number;              // Corresponds to BigDecimal in Java
    bd_bank_nm: string;
    bd_no: string;
    bd_date: Date;               // Corresponds to Date in Java
    bd_amt: number;              // Corresponds to BigDecimal in Java
    mo_rm_no: string;
    mo_date: Date;               // Corresponds to Date in Java
    mo_payer_nm: string;
    mo_id_no: string;
    mo_contact_no: string;
    mo_amt: number;              // Corresponds to BigDecimal in Java
    detail_type: string;
    n_detail_type: string;
    id: number;
    otc_id: number;

    totalFromEMV: number;
}